ydb/core/persqueue/partition.cpp

```diff
@@ -1508,19 +1520,15 @@ void TPartition::AddUserAct(TSimpleSharedPtr<TEvPQ::TEvSetClientInfo> act)
 
 void TPartition::RemoveImmediateTx()
 {
-    Y_ABORT_UNLESS(!UserActionAndTransactionEvents.empty());
-    Y_ABORT_UNLESS(UserActionAndTransactionEventsFrontIs<TEvPersQueue::TEvProposeTransaction>());
+    EnsureWhatUserActionAndTransactionEventsFrontIs<TSimpleSharedPtr<TEvPersQueue::TEvProposeTransaction>>();
 
     UserActionAndTransactionEvents.pop_front();
     --ImmediateTxCount;
 }
 
 void TPartition::RemoveUserAct()
 {
-    Y_ABORT_UNLESS(!UserActionAndTransactionEvents.empty());
-    Y_ABORT_UNLESS(UserActionAndTransactionEventsFrontIs<TEvPQ::TEvSetClientInfo>());
-
```

Comment:
Why don't we need to EnsureWhatUserActionAndTransactionEventsFrontIs?

Reply:
because GetCurrentEvent will check the type of the object at the head of the queue